operator/text/v0/chunk_text.go

```diff
@@ -0,0 +1,137 @@
+package text
+
+import (
+	"fmt"
+
+	"github.com/instill-ai/component/external/langchaingo/textsplitter"
+	"github.com/pkoukk/tiktoken-go"
+)
+
+type ChunkTextInput struct {
+	Text     string   `json:"text"`
+	Strategy Strategy `json:"strategy"`
+}
+
+type Strategy struct {
+	Setting Setting `json:"setting"`
+}
+
+type Setting struct {
+	ChunkMethod       string   `json:"chunk_method,omitempty"`
+	ChunkSize         int      `json:"chunk_size,omitempty"`
+	ChunkOverlap      int      `json:"chunk_overlap,omitempty"`
+	ModelName         string   `json:"model_name,omitempty"`
+	EncodingName      string   `json:"encoding_name,omitempty"`
+	AllowedSpecial    []string `json:"allowed_special,omitempty"`
+	DisallowedSpecial []string `json:"disallowed_special,omitempty"`
+	Separators        []string `json:"separators,omitempty"`
+	KeepSeparator     bool     `json:"keep_separator,omitempty"`
+	CodeBlocks        bool     `json:"code_blocks,omitempty"`
+	ReferenceLinks    bool     `json:"reference_links,omitempty"`
+	// TODO: Add SecondSplitter, which is to set the details about how to chunk the paragraphs in Markdown format.
+	// https://pkg.go.dev/github.com/tmc/langchaingo@v0.1.10/textsplitter#MarkdownTextSplitter
+	// secondSplitter textsplitter.TextSplitter
+}
+
+type ChunkTextOutput struct {
+	ChunkNum   int         `json:"chunk_num"`
+	TextChunks []TextChunk `json:"text_chunks"`
+	TokenCount int         `json:"token_count,omitempty"`
+}
+
+type TextChunk struct {
+	Text          string `json:"text"`
+	StartPosition int    `json:"start_position,omitempty"`
+	EndPosition   int    `json:"end_position,omitempty"`
+}
+
+func (s *Setting) SetDefault() {
+	if s.ChunkSize == 0 {
+		s.ChunkSize = 512
+	}
+	if s.ChunkOverlap == 0 {
+		s.ChunkOverlap = 100
+	}
+	if s.ModelName == "" {
+		s.ModelName = "gpt-3.5-turbo"
+	}
+	if s.EncodingName == "" {
+		s.EncodingName = "cl100k_base"
+	}
+	if s.AllowedSpecial == nil {
+		s.AllowedSpecial = []string{}
+	}
+	if s.DisallowedSpecial == nil {
+		s.DisallowedSpecial = []string{"all"}
+	}
+	if s.Separators == nil {
+		s.Separators = []string{"\n\n", "\n", " ", ""}
+	}
+}
+
+func chunkText(input ChunkTextInput) (ChunkTextOutput, error) {
+	var split textsplitter.TextSplitter
+	setting := input.Strategy.Setting
+	setting.SetDefault()
```

Comment:
What's the point in having settings as user input (e.g. allowed special) if they're going to ve overriden? Shouldn't the `default` field in the config schema determine the default behaviour, rather than the Go code?

Reply:
> Shouldn't the default field in the config schema determine the default behaviour, rather than the Go code?

I also think so. 
However, there is an unknown bug that I could not fix. 
I also feel sad about it. 😢 
- [Ref](https://instill-ai.slack.com/archives/C016F66FX9Q/p1716909862447969?thread_ts=1716560732.412659&cid=C016F66FX9Q)

But, in a positive way, it decreases the risk of the bugs come from other places.

Reply:
Sorry, I reviewed without having all the context in mind. If this is a workaround to a bug that we can't address in this sprint, I'd add a TODO comment before this line explaining why it's needed and how / when it should be fixed.